### PR TITLE
Multi node support and serialization

### PR DESCRIPTION
**Pull request by SebastianSchmidl**

### CODE REVIEW

File: distod/src/main/scala/com/github/codelionx/distod/Serialization.scala [hunk 1]
```diff
@@ -13,4 +24,72 @@ object Serialization {
...
+  class CandidateSetDeserializer extends StdDeserializer[CandidateSet](classOf[CandidateSet]) {
+
+    override def deserialize(p: JsonParser, ctxt: DeserializationContext): CandidateSet =
+      if (p.currentToken() == JsonToken.START_ARRAY) {
+        p.nextToken()
+        val longs = mutable.ArrayBuilder.make[Long]
+        do {
+          val item = _parseLongPrimitive(p, ctxt)
+          longs.addOne(item)
+        } while (p.nextToken() != null && p.currentToken() != JsonToken.END_ARRAY)
```
Comment (CodeLionX) [inline distod/src/main/scala/com/github/codelionx/distod/Serialization.scala]: Codacy Issue found: [Avoid using while loops](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4674182)

File: distod/src/test/scala/com/github/codelionx/distod/SerializationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        CandidateSet.empty,
+        CandidateSet.from(0, 5, 8, 2, 4),
+        CandidateSet.fromSpecific(0 until 200 by 5)
+      )
+
+      testSubjects.foreach(testSubject =>
+        noException should be thrownBy {
+          println(testSubject.toBitMask.map(_.toBinaryString).mkString(" "))
+          // turn it into bytes
+          val bytes = serialization.serialize(testSubject).get
```
Comment (CodeLionX) [inline distod/src/test/scala/com/github/codelionx/distod/SerializationSpec.scala]: Codacy Issue found: [Usage of get on optional type.](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4674182)

File: distod/src/test/scala/com/github/codelionx/distod/SerializationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+          // turn it back into an object
+          val back = serialization.deserialize(bytes, classOf[CandidateSet]).get
```
Comment (CodeLionX) [inline distod/src/test/scala/com/github/codelionx/distod/SerializationSpec.scala]: Codacy Issue found: [Usage of get on optional type.](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4674182)

File: distod/src/main/scala/com/github/codelionx/distod/Serialization.scala [hunk 4]
```diff
@@ -13,4 +24,73 @@ object Serialization {
...
+
+    override def deserialize(p: JsonParser, ctxt: DeserializationContext): CandidateSet =
+      if (p.hasToken(JsonToken.START_ARRAY)) {
+        p.nextToken()
+        // optimization using mutable collection and while loop
+        val longs = mutable.ArrayBuilder.make[Long]
+        do {
+          val item = _parseLongPrimitive(p, ctxt)
+          longs.addOne(item)
+        } while (p.nextToken() != null && !p.hasToken(JsonToken.END_ARRAY))
```
Comment (SebastianSchmidl) [inline distod/src/main/scala/com/github/codelionx/distod/Serialization.scala:59]: Codacy Issue found: [Avoid using while loops](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4674182)

**Pull request merged**